godal.go

```diff
@@ -3419,6 +3419,28 @@ type CloseResultSetOption interface {
 	setReleaseResultSetOpt(rrso *closeResultSetOpts)
 }
 
+func clearUnhandledError() {
+	C.CPLErrorReset()
+}
+
+// getLastUnhandledError returns an error not handled through GDAL's standard error handling interface iff such an error exists
+func getLastUnhandledError() error {
+	errNo := int(C.CPLGetLastErrorNo())
+	errMsgP := C.CPLGetLastErrorMsg()
+
+	var errMsg string
+	if errMsgP != nil {
+		errMsg = C.GoString(errMsgP)
+	}
+
+	if errNo != 0 || errMsg != "" {
+		clearUnhandledError()
+		return fmt.Errorf("unhandled error code %d: %s", errNo, errMsg)
+	}
+
+	return nil
+}
+
```

Comment:
I'm not sure what you are trying to do here? You should make sure that no error is left unhandled, not reset it like this

Reply:
@tbonfort thanks for pointing me in the right direction ! I had completely missed that I needed the backing c++ functions and was confused as to why errors were not being returned.